great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py

```diff
@@ -25,145 +28,271 @@ class ColumnDescriptiveMetricsMetricRetriever(MetricRetriever):
         for metric_name in column_metric_names:
-            if metric_name == "column_values.null.count":
-                value_type = int
             for column in column_list:
-                metric_lookup_key = (metric_name, f"column={column}", tuple())
-                metrics.append(
-                    ColumnMetric[value_type](  # type: ignore[valid-type]  # Will be refactored in upcoming PR
-                        batch_id=validator.active_batch.id,
+                column_metric_configs.append(
+                    MetricConfiguration(
                         metric_name=metric_name,
-                        column=column,
-                        value=computed_metrics[metric_lookup_key],
-                        exception=None,  # TODO: Pass through a MetricException() if an exception is thrown
+                        metric_domain_kwargs={"column": column},
+                        metric_value_kwargs={},
                     )
                 )
+        return column_metric_configs
 
-        return metrics
+    def _compute_metrics(
+        self, batch_request: BatchRequest, metric_configs: list[MetricConfiguration]
+    ):
+        validator = self._context.get_validator(batch_request=batch_request)
+        # The runtime configuration catch_exceptions is explicitly set to True to catch exceptions
+        # that are thrown when computing metrics. This is so we can capture the error for later
+        # surfacing, and not have the entire metric run fail so that other metrics will still be
+        # computed.
+        # TODO: Get aborted_metrics in addition to computed metrics so they can be plumbed into the MetricException.
+        computed_metrics = validator.compute_metrics(
+            metric_configurations=metric_configs,
+            runtime_configuration={"catch_exceptions": True},
+        )
+        assert isinstance(validator.active_batch, Batch)
+        if not isinstance(validator.active_batch, Batch):
+            raise TypeError(
+                f"validator.active_batch is type {type(validator.active_batch).__name__} instead of type {Batch.__name__}"
+            )
+        batch_id = validator.active_batch.id
+        return batch_id, computed_metrics
+
+    def _get_metric_from_computed_metrics(
+        self,
+        metric_name: str,
+        computed_metrics: dict[_MetricKey, Any],
+        metric_lookup_key: _MetricKey | None = None,
+    ):
+        if metric_lookup_key is None:
+            metric_lookup_key = (
+                metric_name,
+                tuple(),
+                tuple(),
+            )
+        value = None
+        exception = None
+        if metric_lookup_key in computed_metrics:
+            value = computed_metrics[metric_lookup_key]
+        else:
+            exception = MetricException(
+                type="TBD", message="TBD"
+            )  # TODO: Fill in type and message if an exception is thrown
```

Comment:
Here is where we are adding a placeholder exception if the metric cannot be found in computed_metrics.